File: LearnAndroid/Day5/app/src/main/java/com/example/day5/Activities/MainActivity.java
package com.example.day5.Activities;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.viewpager.widget.ViewPager;

import android.content.Context;
import android.os.Bundle;
import android.util.AttributeSet;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;

import com.example.day5.Adapter.VPAdapter;
import com.example.day5.CustomNav;
import com.example.day5.Fragment.ClassFragment;
import com.example.day5.Fragment.ClassFragment1;
import com.example.day5.Fragment.ClassFragment2;
import com.example.day5.Fragment.ClassFragment3;
import com.example.day5.Fragment.ClassFragment4;
import com.example.day5.Fragment.HelpCenterFragment;
import com.example.day5.Fragment.HomeFragment;
import com.example.day5.Fragment.LoginFragment;
import com.example.day5.R;
import com.google.android.material.bottomnavigation.BottomNavigationView;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity implements ViewPager.OnPageChangeListener, BottomNavigationView.OnNavigationItemSelectedListener {
    ViewPager vp;
    CustomNav nav;
    List<Fragment> pages;
    VPAdapter adapter;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initView();
        initData();
        initEvent();
    }

    private void initView() {
        vp = findViewById(R.id.vp);
        nav = findViewById(R.id.nav);
    }

    private void initData() {
        pages = new ArrayList<>();
        pages.add(new HomeFragment());
        pages.add(new LoginFragment());
        pages.add(new ClassFragment());
        pages.add(new ClassFragment1());
        pages.add(new ClassFragment2());
        pages.add(new ClassFragment3());
        pages.add(new ClassFragment4());
        pages.add(new HelpCenterFragment());
        adapter = new VPAdapter(getSupportFragmentManager(), pages);
    }

    private void initEvent() {
        vp.setAdapter(adapter);
        vp.addOnPageChangeListener(this);
        nav.setOnNavigationItemSelectedListener(this);
    }

    @Override
    public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {

    }

    @Override
    public void onPageSelected(int position) {
        nav.getMenu().getItem(position).setChecked(true);
    }

    @Override
    public void onPageScrollStateChanged(int state) {

    }

    @Override
    public boolean onNavigationItemSelected(@NonNull MenuItem item) {
        // set getorder 没用
        int id = item.getItemId();
        if (id == R.id.home) {
            vp.setCurrentItem(0);
        } else if (id == R.id.person) {
            vp.setCurrentItem(1);
        } else if (id == R.id.be_on_class) {
            vp.setCurrentItem(2);
        } else if (id == R.id.be_on_class1) {
            vp.setCurrentItem(3);
        } else if (id == R.id.be_on_class2) {
            vp.setCurrentItem(4);
        } else if (id == R.id.be_on_class3) {
            vp.setCurrentItem(5);
        } else if (id == R.id.be_on_class4) {
            vp.setCurrentItem(6);
        } else if (id == R.id.help_center) {
            vp.setCurrentItem(7);
        }
        return false;
    }

}

File: LearnAndroid/Day11/app/src/main/java/com/example/day11/activity/MainActivity.kt
package com.example.day11.activity

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.MenuItem
import android.view.View
import android.widget.FrameLayout
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import androidx.viewpager.widget.ViewPager
import com.example.day11.R
import com.example.day11.RVAdapter
import com.example.day11.adapter.VPAdapter
import com.example.day11.bean.Item
import com.example.day11.fragment.HomeFragment
import com.example.day11.fragment.PersonalFragment
import com.google.android.material.bottomnavigation.BottomNavigationView

class MainActivity : AppCompatActivity(), View.OnClickListener,
    BottomNavigationView.OnNavigationItemSelectedListener, ViewPager.OnPageChangeListener {

    private lateinit var bottom_nav:BottomNavigationView
    private lateinit var vp:ViewPager
    private lateinit var adapter: VPAdapter
    private lateinit var pages:MutableList<Fragment>
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initView()
        initEvent()
    }
    private fun initView() {
        pages = ArrayList()
        pages.add(HomeFragment(this))
        pages.add(PersonalFragment())
        bottom_nav = findViewById(R.id.bottom_nav)
        vp = findViewById(R.id.vp)
        adapter = VPAdapter(pages, supportFragmentManager)
        vp.adapter = adapter
    }
    private fun initEvent() {
        bottom_nav.setOnNavigationItemSelectedListener(this)
        vp.setOnPageChangeListener(this)
    }

    override fun onClick(v: View?) {

    }

    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            R.id.home -> {
                vp.currentItem = 0
                true
            }

            R.id.personal -> {
                vp.currentItem = 1
                true
            }
            // 添加其他menu项的case
            else -> false
        }
    }

    override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {
    }

    override fun onPageSelected(position: Int) {
        bottom_nav.selectedItemId = when (position) {
            0 -> R.id.home
            1 -> R.id.personal
            else -> 0 // 默认返回第一个
        }
    }

    override fun onPageScrollStateChanged(state: Int) {
    }

}
